src/base/net/smtp.cpp

```diff
@@ -89,6 +88,17 @@ namespace
 
         return hostname.toLocal8Bit();
     }
+
+    bool canEncodeAsLatin1(const QStringView string)
```

Comment:
Despite the old code, I think this function should check for ascii char set.
https://en.wikipedia.org/wiki/Unicode_and_email 

https://en.wikipedia.org/wiki/Base64:
>Base64 is also widely used for sending e-mail attachments. This is required because SMTP - in its original form - was designed to transport 7-bit ASCII characters only.

Reply:
>Despite the old code, I think this function should check for ascii char set.

I only intended to get rid of the deprecated class. Someone can improve the existing code if they want to.